File: src/main/java/me/nylestroke/nylemod/block/custom/GoldenStandBlock.java
package me.nylestroke.nylemod.block.custom;

import me.nylestroke.nylemod.entity.client.block.GoldenStandEntity;
import net.minecraft.block.BlockRenderType;
import net.minecraft.block.BlockState;
import net.minecraft.block.BlockWithEntity;
import net.minecraft.block.entity.BlockEntity;
import net.minecraft.util.math.BlockPos;
import org.jetbrains.annotations.Nullable;

public class GoldenStandBlock extends BlockWithEntity {

    public GoldenStandBlock(Settings settings) {
        super(settings);
    }

    @Nullable
    @Override
    public BlockEntity createBlockEntity(BlockPos pos, BlockState state) {
        return new GoldenStandEntity(pos, state);
    }

    @Override
    public BlockRenderType getRenderType(BlockState state) {
        return BlockRenderType.ENTITYBLOCK_ANIMATED;
    }

}

File: src/main/kotlin/me/nylestroke/nylemod/block/custom/GoldenStandBlock.kt
package me.nylestroke.nylemod.block.custom

import me.nylestroke.nylemod.block.entity.GoldenStandEntity
import net.minecraft.block.BlockRenderType
import net.minecraft.block.BlockState
import net.minecraft.block.BlockWithEntity
import net.minecraft.block.entity.BlockEntity
import net.minecraft.util.math.BlockPos

class GoldenStandBlock(settings: Settings) : BlockWithEntity(settings) {
    override fun createBlockEntity(pos: BlockPos, state: BlockState): BlockEntity? {
        return GoldenStandEntity(pos, state)
    }

    override fun getRenderType(state: BlockState): BlockRenderType {
        return BlockRenderType.ENTITYBLOCK_ANIMATED
    }
}